src/server/zset_family.cc

```diff
@@ -1351,6 +1351,74 @@ void ZSetFamily::ZCount(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+vector<ScoredMap> OpDiff(EngineShard* shard, Transaction* t) {
+  ArgSlice keys = t->GetShardArgs(shard->shard_id());
+  DVLOG(1) << "shard:" << shard->shard_id() << ", keys " << vector(keys.begin(), keys.end());
+  DCHECK(!keys.empty());
+
+  vector<ScoredMap> results;
+  results.reserve(keys.size());
+
+  auto& db_slice = shard->db_slice();
+  for (size_t i = 0; i < keys.size(); ++i) {
+    auto it = db_slice.Find(t->GetDbContext(), keys[i], OBJ_ZSET);
+    if (!it) {
+      results.push_back({});
+      continue;
+    }
+
+    ScoredMap sm = FromObject((*it)->second, 1);
+    results.push_back(std::move(sm));
+  }
+
+  return results;
+}
+
+void ZSetFamily::ZDiff(CmdArgList args, ConnectionContext* cntx) {
```

Comment:
General comment here, we can implement `ZDiffStore` ontop of this. This should be a simple PR/follow up and I would be happy to tackle this :)

Reply:
Sounds good :)